src/java/com/rapleaf/jack/queries/where_operators/Between.java

```diff
@@ -1,22 +1,27 @@
 package com.rapleaf.jack.queries.where_operators;
 
-public class Between<V extends Comparable<V>> extends WhereOperator<V> {
+import org.apache.commons.lang.StringUtils;
 
-  public Between(V min, V max) {
-    super(min, max);
-    ensureNoNullParameter();
-  }
+import com.rapleaf.jack.queries.Column;
 
-  @Override
-  public String getSqlStatement() {
-    return "BETWEEN ? AND ?";
-  }
+public class Between<V> extends WhereOperator<V> {
 
-  public V getMin() {
-    return getParameters().get(0);
-  }
+  public Between(V value1, V value2) {
+    super("BETWEEN ? AND ?");
 
-  public V getMax() {
-    return getParameters().get(1);
+    // this operator takes care of its own construction
+    if (!(value1 instanceof Column)) {
```

Comment:
would it work to have a separate constructor for Column arguments?  ex

public Between(Column min, Column max) 

it might be nicer than having the instanceof checks.